### ISSUE TITLE
Gem release latest

### ISSUE THREAD

**Issue by headius** [ruby/matrix#12]
The latest matrix gem is from late 2018 but there have been fixes and enhancements since. JRuby now sources this library from the gem, so we have spec failures based on latest ruby/spec.

We need to get a gem release with the latest sources.

**Comment by headius** [comment]
Failures on latest specs using gem release (on JRuby):

```
174)
Matrix::EigenvalueDecomposition#eigenvalue_matrix returns a diagonal matrix with the eigenvalues on the diagonal ERROR
LoadError: no such file to load -- matrix/eigenvalue_decomposition
/home/travis/build/jruby/jruby/lib/ruby/stdlib/matrix.rb:1446:in `eigensystem'
/home/travis/build/jruby/jruby/spec/ruby/library/matrix/eigenvalue_decomposition/eigenvalue_matrix_spec.rb:6:in `block in <main>'
org/jruby/RubyBasicObject.java:2615:in `instance_eval'
org/jruby/RubyEnumerable.java:1862:in `all?'
org/jruby/RubyArray.java:1809:in `each'
/home/travis/build/jruby/jruby/spec/ruby/library/matrix/eigenvalue_decomposition/eigenvalue_matrix_spec.rb:4:in `<main>'
org/jruby/RubyKernel.java:1009:in `load'
org/jruby/RubyBasicObject.java:2615:in `instance_eval'
org/jruby/RubyArray.java:1809:in `each'

175)
Matrix::EigenvalueDecomposition#eigenvalues returns an array of complex eigenvalues for a rotation matrix ERROR
NameError: uninitialized constant Matrix::EigenvalueDecomposition
Did you mean?  Matrix::EigenvalueDecomposition
org/jruby/RubyModule.java:3810:in `const_missing'
/home/travis/build/jruby/jruby/lib/ruby/stdlib/matrix.rb:1446:in `eigensystem'
/home/travis/build/jruby/jruby/spec/ruby/library/matrix/eigenvalue_decomposition/eigenvalues_spec.rb:6:in `block in <main>'
org/jruby/RubyBasicObject.java:2615:in `instance_eval'
org/jruby/RubyEnumerable.java:1862:in `all?'
org/jruby/RubyArray.java:1809:in `each'
/home/travis/build/jruby/jruby/spec/ruby/library/matrix/eigenvalue_decomposition/eigenvalues_spec.rb:4:in `<main>'
org/jruby/RubyKernel.java:1009:in `load'
org/jruby/RubyBasicObject.java:2615:in `instance_eval'
org/jruby/RubyArray.java:1809:in `each'

...more of the same

188)
Matrix::LUPDecomposition#determinant returns the determinant when the matrix is square ERROR
LoadError: no such file to load -- matrix/lup_decomposition
/home/travis/build/jruby/jruby/lib/ruby/stdlib/matrix.rb:1461:in `lup'
/home/travis/build/jruby/jruby/spec/ruby/library/matrix/lup_decomposition/determinant_spec.rb:7:in `block in <main>'
org/jruby/RubyBasicObject.java:2615:in `instance_eval'
org/jruby/RubyEnumerable.java:1862:in `all?'
org/jruby/RubyArray.java:1809:in `each'
/home/travis/build/jruby/jruby/spec/ruby/library/matrix/lup_decomposition/determinant_spec.rb:4:in `<main>'
org/jruby/RubyKernel.java:1009:in `load'
org/jruby/RubyBasicObject.java:2615:in `instance_eval'
org/jruby/RubyArray.java:1809:in `each'

189)
Matrix::LUPDecomposition#determinant raises an error for rectangular matrices ERROR
NameError: uninitialized constant Matrix::LUPDecomposition
Did you mean?  Matrix::LUPDecomposition
org/jruby/RubyModule.java:3810:in `const_missing'
/home/travis/build/jruby/jruby/lib/ruby/stdlib/matrix.rb:1461:in `lup'
/home/travis/build/jruby/jruby/spec/ruby/library/matrix/lup_decomposition/determinant_spec.rb:15:in `block in <main>'
org/jruby/RubyArray.java:1809:in `each'
/home/travis/build/jruby/jruby/spec/ruby/library/matrix/lup_decomposition/determinant_spec.rb:12:in `block in <main>'
org/jruby/RubyBasicObject.java:2615:in `instance_eval'
org/jruby/RubyEnumerable.java:1862:in `all?'
org/jruby/RubyArray.java:1809:in `each'
/home/travis/build/jruby/jruby/spec/ruby/library/matrix/lup_decomposition/determinant_spec.rb:4:in `<main>'
org/jruby/RubyKernel.java:1009:in `load'
org/jruby/RubyBasicObject.java:2615:in `instance_eval'
org/jruby/RubyArray.java:1809:in `each'

... more of the same
```

**Comment by headius** [comment]
Seems like those files are just not in the gem? 🤔

**Comment by marcandre** [comment]
I finally synchronized the main repo with this one (there were missing commits in both directions). I got version `0.2.0` with the same code by rewriting history a little bit for this repo (although I haven't released a 0.2.0` as I doubt it would do anyone any good).
Since there were some meaningful recent changes, I released `0.3.0` which I hope will work for everyone 😸  (let me know if you encounter any issue).
Thank you for your patience.
